src/frame/headers.rs

```diff
@@ -153,6 +153,9 @@ impl Headers {
 
         // Read the padding length
         if flags.is_padded() {
+            if src.len() < 1 {
+                return Err(Error::MalformedMessage);
+            }
             // TODO: Ensure payload is sized correctly
```

Comment:
Maybe there's more involved, but wanted to check: does this addition essentially complete this TODO?

Reply:
Seems like it; I removed the comment.